repository-ydb-v2/src/main/java/tech/ydb/yoj/repository/ydb/table/YdbTable.java

```diff
@@ -156,7 +177,7 @@ public <ID extends Entity.Id<T>> Stream<ID> streamAllIds(int batchSize) {
     @Override
     public <ID extends Entity.Id<T>> Stream<ID> streamPartialIds(ID partial, int batchSize) {
         Preconditions.checkArgument(1 <= batchSize && batchSize <= 10000, "batchSize must be in range [1, 10000], got %s", batchSize);
-        return StreamSupport.stream(new BatchFindSpliterator<>(type, partial, batchSize) {
+        return StreamSupport.stream(new BatchFindSpliterator<>(descriptor.clazz(), partial, batchSize) {
```

Comment:
Only passing entity class to `BatchFindSpliterator` looked **suspicious**. It is in fact OK: `BatchFindSpliterator` only uses the Entity **ID** schema, never using the table name and never creating `YqlStatement`s by itself.

**But let's clarify this code** by making `BatchFindSpliterator` accept `EntityIdSchema<ID> idSchema` instead of `Class<T> entityType`, and pass the `EntityIdSchema.<ID>ofEntity(descriptor.clazz())` to `BatchFindSpliterator` constructor from `streamPartial[...]()` family of methods. We can do this because `BatchFindSpliterator` **is package-private, not a part of public API!**